### PR TITLE
Abs algo

### PR DESCRIPTION
**Pull request by i-redbyte**

### CODE REVIEW

File: math/binary/abs.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package binary
+
+// ABS - this function return absolute value use binary operation
+func ABS(base, n int) int {
+	m := n >> (base - 1)
```
Comment (raklaptudirm) [inline math/binary/abs.go:5]: Explain these bit-wise operations in comments.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: math/binary/abs.go (hunk 1 above)
Comment (i-redbyte) [reply math/binary/abs.go:10]: Done

File: math/binary/abs.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package binary
+
+// ABS - this function return absolute value use binary operation
+// Principle of operation:
+// 1) Get the mask by right shift by the bit width of the variable (int64, int32, int16, int8)
```
Comment (raklaptudirm) [inline math/binary/abs.go:5]: Since the method accepts an `int`, supporting other lengths is not required in my opinion. It will always be an `int`.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: math/binary/abs.go (hunk 2 above)
Comment (i-redbyte) [reply math/binary/abs.go:5]: > Since the method accepts an `int`, supporting other lengths is not required in my opinion. It will always be an `int`.

this is added only to describe the algorithm, to understand that the shift must be a multiple of the digit of your integer

Comment (raklaptudirm) [reply math/binary/abs.go:5]: In my opinion, it can be adequately described in the comments, and a separate variable is redundant.

Comment (i-redbyte) [reply math/binary/abs.go]: done

File: math/abs/abs.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// description: Absolute value
+// details:
+// In mathematics, the absolute value or modulus of a real number x, denoted |x|, is the non-negative value of x without regard to its sign. [Absolute value](https://en.wikipedia.org/wiki/Average#Arithmetic_mean)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see abs_test.go
+
+package abs
+
+// ABS - this function return absolute value
+func ABS(n int) int {
```
Comment (tjgurwara99) [inline math/abs/abs.go:11]: Why use `ABS` instead of `Abs`??

```suggestion
func Abs(n int) int {
```

Comment (tjgurwara99) [reply math/abs/abs.go:11]: Also, this should just exist in the math package not in a separate package.

File: math/binary/abs.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+// 2) For negative numbers, above step sets mask as 1 1 1 1 1 1 1 1 and 0 0 0 0 0 0 0 0 for positive numbers.
+// 3) Add the mask to the given number.
+// 4) XOR of mask + n and mask gives the absolute value.
+func ABS(base, n int) int {
```
Comment (tjgurwara99) [inline math/binary/abs.go:10]:
```suggestion
func Abs(base, n int) int {
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: math/abs/abs.go (hunk 3 above)
Comment (i-redbyte) [reply math/abs/abs.go]: Done

File: math/binary/abs.go (hunk 4 above)
Comment (i-redbyte) [reply math/binary/abs.go]: Done

File: math/binary/abs.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package binary
+
+// Abs - this function return absolute value use binary operation
```
Comment (tjgurwara99) [inline math/binary/abs.go:3]:
```suggestion
// Abs returns absolute value use binary operation
```

File: math/abs.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// abs.go
+// description: Absolute value
+// details:
+// In mathematics, the absolute value or modulus of a real number x, denoted |x|, is the non-negative value of x without regard to its sign. [Absolute value](https://en.wikipedia.org/wiki/Average#Arithmetic_mean)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see abs_test.go
+
+package math
+
+// Abs - this function return absolute value
```
Comment (tjgurwara99) [inline math/abs.go:10]:
```suggestion
// Abs returns absolute value
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Small changes.

File: math/abs_test.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
+package math
+
+import (
+	"github.com/TheAlgorithms/Go/math/binary"
+	"math"
+	"testing"
+)
+
+func TestABS(t *testing.T) {
```
Comment (tjgurwara99) [inline math/abs_test.go:9]:
```suggestion
func TestAbs(t *testing.T) {
```

File: math/abs_test.go [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+func TestBinaryABS(t *testing.T) {
+	tests := getTests()
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := binary.Abs(64, test.n); got != test.want {
+				t.Errorf("Abs() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
```
Comment (tjgurwara99) [inline math/abs_test.go:29]: Why did you move this here when `abs.Binary` is not in the package? This does not belong in this package. Move it inside the binary package.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: math/abs_test.go (hunk 7 above)
Comment (i-redbyte) [reply math/abs_test.go]: Done. is everything ok now?

Comment (i-redbyte) [reply math/abs_test.go]: Done

File: math/abs_test.go (hunk 8 above)
Comment (i-redbyte) [reply math/abs_test.go]: Done. is everything ok now?

**Review by tjgurwara99 — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]
